client/my-sites/plugins/plugin-list-header/index.jsx

```diff
@@ -166,10 +166,20 @@ export default React.createClass( {
 
 		if ( ! this.props.isBulkManagementActive ) {
 			if ( 0 < this.props.pluginUpdateCount ) {
+				var buttonText = this.translate( 'Update', { context: 'button label' } );
+				
+				buttonText += ' ' + this.numberFormat( this.props.pluginUpdateCount ) + ' ';
+
+				if ( this.props.pluginUpdateCount > 1 ) {
+					buttonText += this.translate( 'Plugins', { context: 'button label' } );
+				} else {
+					buttonText += this.translate( 'Plugin', { context: 'button label' } );
+				}
+
 				rightSideButtons.push(
 					<ButtonGroup key="plugin-list-header__buttons-update-all">
 						<Button compact primary onClick={ this.props.updateAllPlugins } >
-							{ this.translate( 'Update All', { context: 'button label' } ) }
+							{ buttonText }
```

Comment:
I think you could simplify using [pluralization](https://github.com/Automattic/i18n-calypso#pluralization), something like ...

```es6
i18n.translate(
    'Update Plugin.',
    'Update %(numberOfPlugins) plugins.',
    {
        count: this.props.pluginUpdateCount,
        args: {
            numberOfPlugins: this.props.pluginUpdateCount
        }
    }
);
```

On other hand I think it isn;t needed add format to plugins number since it's an Integer.